examples/cms-contentful/app/api/draft/route.ts

```diff
@@ -0,0 +1,22 @@
+import { draftMode } from 'next/headers'
+import { redirect } from 'next/navigation'
+import { getPreviewPostBySlug } from '../../../lib/api'
+
+export async function GET(request: Request) {
+  const { searchParams } = new URL(request.url)
+  const secret = searchParams.get('secret')
+  const slug = searchParams.get('slug')
+
+  if (secret !== process.env.CONTENTFUL_PREVIEW_SECRET || !slug) {
```

Comment:
```suggestion
  if (secret !== process.env.CONTENTFUL_PREVIEW_SECRET) {
```

This is checking for `!slug` but the error says "invalid token"

Seems like line 16 and 17 will handle that case instead.

Reply:
Good catch!